crates/swc_ecma_transforms_compat/src/es2017/async_to_generator.rs

```diff
@@ -246,14 +244,15 @@ impl Fold for Actual {
                     _ => {}
                 }
 
-                let assign_expr = assign_expr.fold_children_with(self);
+                assign_expr.visit_mut_children_with(self);
                 self.in_prototype_assignment = false;
-                return Expr::Assign(assign_expr);
+                *expr = Expr::Assign(assign_expr.take());
```

Comment:
This is redundant as `expr` is always `Expr::Assign(assign_expr)`.
You can just delete this line.